torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,290 @@
+        "mid_latitude_north": "0f8382ca6554fb7cf9aff42226a14f9d",
+        "mid_latitude_south": "b17cf6d023f752c533211fdb742f296b",
+        "polar_north": "73923dcaf1b9b79bad82de1aa0740d1e",
+        "polar_south": "3bc9f4c6f8955b10b4d55d23e0ab2da7",
+        "subtropical_north": "f8f039970256902e6e9ebd6747589294",
+        "subtropical_south": "8346d73a983396c5d41b577c3a94bc26",
+        "tropical": "abe19b22b5d031e6b609cc7207706c3d",
+    }
+
+    classes = ["Fill", "Cloud Shadow", "Clear", "Thin Cloud", "Cloud"]
+
+    filename_glob = "L7*.TIF"
+
+    filename_regex = r"""
+        ^L7[12]
+        (?P<wrs_path>\d{3})
+        (?P<wrs_row>\d{3})
+        _(?P=wrs_row)
+        (?P<date>\d{8})
+        _(?P<band>B\d{2})
+        \.TIF$
+    """
+    date_format = "%Y%m%d"
+
+    separate_files = True
+    rgb_bands = ["B40", "B20", "B30"]
+    all_bands = ["B10", "B20", "B30", "B40", "B50", "B61", "B62", "B70", "B80"]
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        bands: Sequence[str] = all_bands,
+        cache: bool = True,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Landsat7 Cloud Cover Assessment Validation dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            crs: :term:`coordinate reference system (CRS)` to warp to
+            (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+            (defaults to the resolution of the first file found)
+            bands: bands to return
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        self.root = root
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        super().__init__(
+            root, crs=crs, res=res, bands=bands, transforms=transforms, cache=cache
+        )
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        pathname = os.path.join(self.root, "**", self.filename_glob)
+        for fname in glob.iglob(pathname, recursive=True):
+            if not fname.endswith(".tar.gz"):
```

Comment:
This if-statement is unnecessary, the glob already prevents matching files ending in .tar.gz